Sources/Commands/Workspace.swift

```diff
@@ -545,7 +532,9 @@ public class Workspace {
         data["version"] = .int(Workspace.currentSchemaVersion)
         data["dependencies"] = .array(dependencies.map{ $0.toJSON() })
 
-        // FIXME: This should write atomically.
-        try localFileSystem.writeFileContents(statePath, bytes: JSON.dictionary(data).toBytes())
+        // Write atomically.
+        let tempFile = try TemporaryFile()
+        try localFileSystem.writeFileContents(tempFile.path, bytes: JSON.dictionary(data).toBytes())
+        try rename(old: tempFile.path.asString, new: statePath.asString)
```

Comment:
I think it is more likely it should move to part of the `FileSystem` API (as Foundation does). Correctly writing atomically is a very file system specific operation (the approach here actually isn't 100% correct on macOS in the case of immediate power loss, but the correct approach is much more expensive).

I would probably rather not fix this here (the risk of this bug being a major problem in the short term is low) and only try to fix it the "right" way.